skimage/morphology/extrema.py

```diff
@@ -275,10 +279,11 @@ def local_maxima(image, selem=None):
 
     The resulting image will contain all 6 local maxima.
     """
-    if np.issubdtype(image.dtype, 'half'):
-        # find the minimal grey level difference
-        h = _find_min_diff(image)
-    else:
+    # find the minimal grey level difference
+    h = _find_min_diff(image)
+    if h == 0:
+        return np.zeros(image.shape, np.uint8)
```

Comment:
`np.zeros(image.shape, np.uint8)` is in accordance with the convention set by `h_minima` and `h_maxima` in which the dtype of the returned matrix is always `np.uint8`.